azuremetadata/lib/AzureMetadata/AzureGeneral.pm

```diff
@@ -20,13 +20,71 @@ use warnings;
 use XML::LibXML;
 
 our @ISA    = qw (Exporter);
-our @EXPORT_OK = qw (get_instance_id);
+our @EXPORT_OK = qw (
+  get_instance_name
+  get_instance_id
+);
 
 sub get_instance_id {
+  my $id = fixup_guid_endianness(`dmidecode | grep UUID | cut -c8-`);
```

Comment:
Should not execute a system call as argument to a function. This provides no opportunity to deal with any potential errors.